src/backend/aspen/workflows/transform_gisaid/update_locations.py

```diff
@@ -0,0 +1,151 @@
+    to instead be a GISAID -> CZ GEN EPI locations mapping file. Some of the locations
+    in our database don't follow the Nextstrain location conventions and upstream
+    changes to Nextstrain's translations have broken our tree builds in the past.
+
+    This translation file is tab-delimited with one source and destination mapping per line:
+    old_location\tnew_location
+
+    There are 4 main changes we make to the file:
+    1. There are a few localities we work with that don't want their location rolled
+       into a larger area (ex: city into county). Remove these localities from the
+       output file.
+
+    2. Remove any rules that translate a location in use by CZ GEN EPI into some other
+       name.
+
+       For example, Nexstrain may have a rule like this:
+       North America/USA/New York/Orange County    North America/USA/New York/Orange County NY
+
+       If we have a group that's using the "New York/Orange County" location, we want
+       to make sure their samples (and any GISAID samples from Orange County) are
+       displayed on a tree with the same location name.
+
+    3. For any translations *to* a location that were skipped in #2 above, we need to
+       change the new location so it matches the value in use by CZ GEN EPI. For example,
+       Nextstrain often rolls up several smaller locations into a larger one like this:
+       North America/USA/New York/Middletown    North America/USA/New York/Orange County NY
+
+       But since we skipped mapping "Orange County" to "Orange County NY" above, we want
+       to update this rule to look like this:
+       North America/USA/New York/Middletown    North America/USA/New York/Orange County
+
+    3. Add a few custom mappings of our own.
+    """
+    update_locations(input_fh, output_fh)
+
+
+def update_locations(input_fh: io.TextIOBase, output_fh: io.TextIOBase):
+    interface: SqlAlchemyInterface = init_db(get_db_uri(Config()))
+    skip_translations = set([])  # Any translations we want to drop *without* remapping
+    with session_scope(interface) as session:
+        remap_translations = get_excluded_locations(session)
+
+    # Where we stash output rows. The file is ~3MB so memory isn't an issue
+    outputs = []
+
+    # Keep track of which rules we've removed so we can remap any
+    # destinations as necessary.
+    translate_destinations = {}
+
+    # Locations to skip entirely without remapping.
+    skip_translations.add("North America/USA/Illinois/Chicago")
+
+    reader = csv.reader(input_fh, delimiter="\t")
+    # Load our file into memory and generate a map of destinations that
+    # need to be rewritten
+    for row in reader:
+        source, dest = row
+        # NOTE: The order of these checks matters!
+        if source in skip_translations:
+            continue
+        if source in remap_translations:
+            translate_destinations[dest] = source
+            continue
```

Comment:
1. so this continue says, don't do the outputs.append and go to the next row, right?

Reply:
Yes, it skips the rest of the stuff in the `for` loop and goes to the next line in the file